src/interfaces/chain.cpp

```diff
@@ -325,7 +325,11 @@ class ChainImpl : public Chain
     CFeeRate relayMinFee() override { return ::minRelayTxFee; }
     CFeeRate relayIncrementalFee() override { return ::incrementalRelayFee; }
     CFeeRate relayDustFee() override { return ::dustRelayFee; }
-    bool getPruneMode() override { return ::fPruneMode; }
+    bool havePruned() override
+    {
+        LOCK(cs_main);
```

Comment:
I'm not suggesting you change this PR, but for future reference we now have a handy `WITH_LOCK` macro: https://github.com/bitcoin/bitcoin/blob/fd61b9fc22faaa946ca43211c6c43b40ca4a860c/src/sync.h#L209

Reply:
I fail to see how this makes the code any easier to read